sdk/core/perf/src/base_test.cpp

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/perf/base_test.hpp"
+#include "azure/core/http/policies/policy.hpp"
+#include "azure/core/internal/http/pipeline.hpp"
+
+#include <functional>
+#include <string>
+#include <vector>
+
+using namespace Azure::Core::Http;
+using namespace Azure::Core::Http::Policies;
+using namespace Azure::Core;
+namespace {
+
+class ProxyPolicy final : public HttpPolicy {
+private:
+  Azure::Perf::BaseTest* m_testContext;
+
+public:
+  ProxyPolicy(Azure::Perf::BaseTest* proxyManager) : m_testContext(proxyManager) {}
+
+  // copy
+  ProxyPolicy(ProxyPolicy const& other) : ProxyPolicy{other.m_testContext} {}
+
+  // move
+  ProxyPolicy(ProxyPolicy&& other) : m_testContext{other.m_testContext} {}
+
+  std::unique_ptr<RawResponse> Send(
+      Request& request,
+      NextHttpPolicy nextPolicy,
+      Context const& context) const override
+  {
+    std::string const recordId(m_testContext->m_recordId);
+    if (recordId.empty())
+    {
+      return nextPolicy.Send(request, context);
+    }
+
+    // Use a new request to redirect
+    auto redirectRequest = Azure::Core::Http::Request(
+        request.GetMethod(), Azure::Core::Url(m_testContext->m_proxy), request.GetBodyStream());
+    if (!request.ShouldBufferResponse())
+    {
+      // This is a download with keep connection open. Let's switch the request
+      redirectRequest = Azure::Core::Http::Request(
+          request.GetMethod(), Azure::Core::Url(m_testContext->m_proxy), false);
+    }
```

Comment:
c++ doesn't support creating a variable with no type `auto variable;`  (I miss JS when I see that :)

The only alternative would be to create a pointer to a request as `nullptr` and then create the request, but that would use Heap instead of Stack to place the request, which at the end is lees performance :(

The only time we'll go inside the `!ShouldBufferResponse()` is for Download api for Storage.